v1_19/src/main/java/com/denizenscript/denizen/nms/v1_19/helpers/ItemHelperImpl.java

```diff
@@ -168,12 +168,10 @@ public void registerStonecuttingRecipe(String keyName, String group, ItemStack r
     }
 
     @Override
-    public void registerSmithingRecipe(String keyName, ItemStack result, ItemStack[] baseItem, boolean baseExact, ItemStack[] upgradeItem, boolean upgradeExact) {
-        ResourceLocation key = new ResourceLocation("denizen", keyName);
-        Ingredient baseItemRecipe = itemArrayToRecipe(baseItem, baseExact);
-        Ingredient upgradeItemRecipe = itemArrayToRecipe(upgradeItem, upgradeExact);
-        LegacyUpgradeRecipe recipe = new LegacyUpgradeRecipe(key, baseItemRecipe, upgradeItemRecipe, CraftItemStack.asNMSCopy(result)); // TODO: 1.19.4: smithing system was changed
-        ((CraftServer) Bukkit.getServer()).getServer().getRecipeManager().addRecipe(recipe);
+    public void registerSmithingRecipe(String keyName, ItemStack result, RecipeChoice baseItem, RecipeChoice upgradeItem, RecipeChoice template) {
+        NamespacedKey key = new NamespacedKey("denizen", keyName);
+        org.bukkit.inventory.SmithingRecipe recipe = new org.bukkit.inventory.SmithingRecipe(key, result, baseItem, upgradeItem);

```

Comment:
There are two possible scenarios:
A: This code is not NMS specific (as yours isn't) and shouldn't be in an NMS module
B: There was a reason all the recipe code was in an NMS module and ignoring the bukkit recipe tools, and you need to keep it that way.

I know for a fact there was a reason, however I do not remember what that reason was. I would advise significant caution changing this facet of the code until/unless you're sure both (1) what the original reason was and (2) that it no longer applies.